src/lamoda/routers.py

```diff
@@ -1,33 +1,144 @@
+CacheMngr = Annotated[RedisCacheManager, Depends(get_cache_manager)]
+settings = LamodaSettings()
 
 
 @lamoda_router.post("/product")
-def parse_product(url: str, collection: LamodaCollection):
-    """View for parsing product by its url"""
+async def parse_product(url: str, db: LamodaDb, cache: CacheMngr):
+    """View for parsing product by its url, processed product is saved to cache and db"""
 
+    validate_url(url, is_product=True, is_category=False)
+    params = {}
+    key_for_cache = {"url": url, "params": params}
+    object_from_cache = await cache.get_object_from_cache(key_for_cache)
+    if object_from_cache:
+        if object_from_cache["status"] == ObjectStatus.PROCESSED.name:
+            return {"message": "object is already processed"}
     product = parse_object(url)
-    Decimal128(str(product.price))
-    product_as_dict = product.dict()
-    for key, value in product_as_dict.items():
-        if isinstance(value, Decimal):
-            product_as_dict[key] = str(value)
-    product.price = str(product.price)
-    created_id = collection.insert_one(product_as_dict)  # it is temporary, made to check the ability to save into db
-    created = collection.find_one({"_id": created_id.inserted_id})  # soon will make other alg to save into db
-    return product.dict()
-
-
-@lamoda_router.get("/")
-def main_page():
-    return {"message": "success"}
+    await db.save_one_product(product)
+    await cache.save_to_cache(
+        key_for_cache,
+        60 * 5,
+        LamodaResponseFromParser(
+            url=url, status=ObjectStatus.PROCESSED.name, params=params, data=product
+        ),
+    )
+    return {"message": "processed"}
+
+
+@lamoda_router.post("/category")
+async def parse_category(url: str, db: LamodaDb, cache: CacheMngr):
+    """View for parsing category, processed category is saved to cache and db"""
+
+    validate_url(url, is_product=False, is_category=True)
+    params = {}
+    key_for_cache = {"url": url, "params": params}
+    object_from_cache = await cache.get_object_from_cache(key_for_cache)
+    if object_from_cache:
+        if object_from_cache["status"] == ObjectStatus.PROCESSED.name:
+            return {"message": "object is already processed"}
+    category = parse_lamoda_category(url)
+    category_id = await db.save_one_category(category)
+    created_category = await db.get_one_category(ObjectId(category_id))
+    for product in parse_links_from_category(created_category):
+        await db.save_one_product(product)
+    created_category = await db.get_one_category(ObjectId(category_id))
+    await cache.save_to_cache(
+        key_for_cache,
+        60 * 5,
+        LamodaResponseFromParser(
+            url=url,
+            status=ObjectStatus.PROCESSED.name,
+            params=params,
+            data=created_category,
+        ),
+    )
+    return {"message": "processed"}
+
+
+@lamoda_router.get("/parse/product", response_model=LamodaResponseFromParser)
```

Comment:
it should be the same url as for post

GET (all) /POST: /products
GET (one)/ UPDATE/ DELETE /products/<id>

Reply:
Those endpoints are very different and they are not connected, one is for parsing manualy(without messages to kafka) another is for making a request to kafka for parsing. And when you send get request for /parse/product it basically it calls view from another application with url  /products. Do you think it is better to merge those endpoint to one url?